peekingduck/utils/logger.py

```diff
@@ -19,37 +19,56 @@
 import os
 import sys
 import logging
+import traceback
 from typing import Optional, Dict
 
 from colorama import init, Fore, Style
 
 
-def setup_logger() -> None:
-    """ Universal logging configuration """
+class LoggerSetup:  # pylint: disable=too-few-public-methods
+    """ Set up the universal logging configuration """
 
-    if os.name == 'nt':
-        init()
+    def __init__(self) -> None:
+        if os.name == 'nt':
+            init()
 
-    formatter = ColoredFormatter('{asctime} {name} {level_color} {levelname}'
-                                 '{reset}: {msg_color} {message} {reset}',
-                                 style='{', datefmt='%Y-%m-%d %H:%M:%S',
-                                 colors={'DEBUG': Fore.CYAN + Style.BRIGHT,
-                                         'INFO': Fore.GREEN + Style.BRIGHT,
-                                         'WARNING': Fore.YELLOW + Style.BRIGHT,
-                                         'ERROR': Fore.RED + Style.BRIGHT,
-                                         'CRITICAL': Fore.RED + Style.BRIGHT})
+        formatter = ColoredFormatter('{asctime} {name} {level_color} {levelname}'
+                                     '{reset}: {msg_color} {message} {reset}',
+                                     style='{', datefmt='%Y-%m-%d %H:%M:%S',
+                                     colors={'DEBUG': Fore.CYAN + Style.BRIGHT,
+                                             'INFO': Fore.GREEN + Style.BRIGHT,
+                                             'WARNING': Fore.YELLOW + Style.BRIGHT,
+                                             'ERROR': Fore.RED + Style.BRIGHT,
+                                             'CRITICAL': Fore.RED + Style.BRIGHT})
 
-    handler = logging.StreamHandler(sys.stdout)
-    handler.setFormatter(formatter)
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setFormatter(formatter)
 
-    logger = logging.getLogger()
-    logger.handlers[:] = []
-    logger.addHandler(handler)
-    logger.setLevel(logging.INFO)
+        self.logger = logging.getLogger()
+        self.logger.handlers[:] = []
+        self.logger.addHandler(handler)
+        self.logger.setLevel(logging.INFO)
+        sys.excepthook = self.handle_exception
```

Comment:
Due to the way this is called, and us needing to use logger.warn within the handle_exception method, I had to make the logger setup it's own class. The alternative was to have the handle_exception function within the setup_logger function, but we have not been taking that approach so far.